codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameWriter.java

```diff
@@ -56,9 +56,10 @@
      * @param endStream indicates if this is the last frame to be sent for the stream.
      * @param promise the promise for the write.
      * @return the future for the write.
+     * @throws Http2Exception if an exception occurs while encoding headers.
      */
     ChannelFuture writeHeaders(ChannelHandlerContext ctx, int streamId, Http2Headers headers,
-            int padding, boolean endStream, ChannelPromise promise);
+            int padding, boolean endStream, ChannelPromise promise) throws Http2Exception;
```

Comment:
@nmittler - I choose to modify the method signature so implementations are free to throw stream errors or connection errors. This may have impact on user applications, and makes the potential for an exception to be generated explicit. We could apply this rational consistently to other methods in this interface, but currently no other methods have a need to throw.